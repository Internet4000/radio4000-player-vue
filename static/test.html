<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1">
		<title>Test Radio4000 player</title>
		<style>
			/* styles for this demo */
			body {
				margin: 0;
				padding: 2rem;
			}
		</style>
	</head>
	<body>

		<radio4000-player channel-slug="good-time-radio"></radio4000-player>

		<!-- <script src="dist/radio4000-player.js"></script> -->
		<script src="dist/radio4000-player.min.js"></script>

	</body>
</html>
