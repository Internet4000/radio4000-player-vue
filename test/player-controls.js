import test from 'ava'
import Vue from 'vue'
import Component from '../src/PlayerControls.vue'

test.cb('renders', t => {
	const vm = new Vue(Component).$mount()
	const tree = {$el: vm.$el.outerHTML}
	const $ = selector => vm.$el.querySelectorAll(selector)
	const playpauseBtn = $('.PlayerControl-group--large')[0].querySelector('button')

	t.snapshot(tree)

	t.is($('.Btn--mute').length, 1)
	t.is($('.Btn--shuffle').length, 1)

	t.is(playpauseBtn.textContent.trim(), 'Play')
	vm.$props.isPlaying = true
	vm.$nextTick(() => {
		t.is(playpauseBtn.textContent.trim(), 'Pause')
		t.end()
	})
})
