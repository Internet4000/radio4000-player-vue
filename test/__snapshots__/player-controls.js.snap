// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "$el": "<menu><div class=\\"PlayerControl-group\\"><button title=\\"Mute on/off\\" class=\\"Btn Btn--mute\\"><span></span></button></div><div class=\\"PlayerControl-group\\"><button title=\\"Shuffle on/off\\" class=\\"Btn Btn--shuffle is-active\\"><span>‚Üù</span></button></div><div class=\\"PlayerControl-group PlayerControl-group--large\\"><div class=\\"PlayPause Btn\\"><input type=\\"checkbox\\" id=\\"id1\\" class=\\"PlayPause-state\\"><label for=\\"id1\\" title=\\"Toggle playback\\" class=\\"PlayPause-label\\"><div class=\\"PlayPause-controller\\"></div></label></div></div><div class=\\"PlayerControl-group PlayerControl-group--large\\"><button title=\\"Next track\\" class=\\"Btn\\">Next</button></div></menu>",
}
`;
